Source/Bogus.Tests/DataSetTests/DateTest.cs

```diff
@@ -352,5 +353,236 @@ public void can_get_timezone_string()
       {
          date.TimeZoneString().Should().Be("Asia/Yerevan");
       }
+
+      public class FactWhenDaylightSavingsSupported : FactAttribute
+      {
+         public FactWhenDaylightSavingsSupported()
+         {
+            if (!TimeZoneInfo.Local.SupportsDaylightSavingTime)
+            {
+               Skip = "Test is only meaningful when Daylight Savings is supported by the local timezone.";
+            }
+         }
+      }
+
+      [FactWhenDaylightSavingsSupported]
+      public void will_not_generate_values_that_do_not_exist_due_to_daylight_savings()
+      {
+         // Arrange
+         var faker = new Faker();
+
+         faker.Random = new Randomizer(localSeed: 5);
+
+         var dstRules = TimeZoneInfo.Local.GetAdjustmentRules();
+
+         var now = DateTime.Now;
+
+         var effectiveRule = dstRules.Single(rule => (rule.DateStart <= now) && (rule.DateEnd >= now));
+
+         var transitionStartTime = CalculateTransitionDateTime(now, effectiveRule.DaylightTransitionStart);
```

Comment:
New to this code base, but what's the consensus on generating test helper methods or classes that can contain boiler plate construction and arrangement of objects?

Reply:
My convention when writing tests has been to make the test methods completely logically independent of one another, but I have seen tests and test suites that reuse a great deal of setup. This could be refactored, but was written this way intentionally (if without a great deal of thought, per se).

Reply:
I would also prefer consolidation of the test initialization, what's your opinion @bchavez?